### PR TITLE
feat: Added bmm tensor instance in frontend paddle

### LINKED ISSUE #26078: bmm

### PR DESCRIPTION
**Pull request by binael**
# PR
Added bmm tensor instance method for paddle frontend and the tests

## Related Issue
None

Close #26078

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### Socials
https://twitter.com/binaelnchekwube

### CODE REVIEW

**Comment by YashGaur000** [comment]
@binael how did u add this bmm function in it? how did u get know what arguments need to be passed in bmm ? i wanna contribute too by making gather instance variable but I ain't getting that what should I pass as arguments and what will I write in its body then also how did u write the tests? plz tell me I wanna do too

**Comment by YashGaur000** [comment]
> > @binael how did u add this bmm function in it? how did u get know what arguments need to be passed in bmm ? i wanna contribute too by making gather instance variable but I ain't getting that what should I pass as arguments and what will I write in its body then also how did u write the tests? plz tell me I wanna do too. You seem to be experienced in it please assist me.
> 
> Please, can you hit me up on discord and let us talk about it "discord username: dael_n" This place is mostly about PR comments, reviews etc

I sent u friend request if u dint get it then hit me up on "kiyotaka6000"

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py [hunk 1]
```diff
@@ -148,6 +148,29 @@ def _get_dtype_and_square_matrix(draw):
     return dtype, mat
 
 
+# bmm helper function
+@st.composite
+def _get_dtype_and_values_bmm(draw):
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:153]: As the input tensor should be a `3-D` tensor we can simply restrict this in `helpers.dtype_and_values` by specifying the `min_num_dims` and `max_num_dims`  as `3` with no need for extra strategy. With this, we can also control the size of each dimension of the tensor.

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 2]
```diff
@@ -828,6 +828,10 @@ def real(self, name=None):
     def cast(self, dtype):
         return paddle_frontend.cast(self, dtype)
 
+    @with_unsupported_dtypes({"2.5.1 and below": ("float16", "bfloat16")}, "paddle")
+    def bmm(self, y, transpose_x=False, transpose_y=False, name=None):
+        return paddle_frontend.bmm(self, y)
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/paddle/tensor/tensor.py:833]: We can add all parameters of the function as indicated in its implementation https://github.com/unifyai/ivy/blob/c01ad72e89914df435981f4d1d5100a76dcbe6e7/ivy/functional/frontends/paddle/linalg.py#L19 😊

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @binael 

Thanks for contributing to Ivy 🤗
Below are comments of a few changes needed
Thanks!

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py (hunk 1 above)
Comment (binael) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:153]: Thank you so much for the swift feedback.
I initially thought of using the above method to:
dtype_and_x=helpers.dtype_and_values(
    available_dtypes=("float"),
    shared_dtype=True,
    num_arrays=2,
    min_num_dims=3,
    max_num_dims=3,
),

Comment (hirwa-nshuti) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:153]: This makes sense, you can stick with current approach

Comment (binael) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:153]: Please, I want to be clear...is it with the helpers.dtype_and_values above or with the (b, m, k) and (b, k, n) previously implemented?

**Comment by binael** [comment]
Hi, 
I noticed that "merging and pushing" to forked repo doesn't update the _two recent details_ of the submodule **"docs/demos/demos_conf.py"** and hence always behind the main repo

I tried manually re-adding the data, and pushing, but it never showed the updates
I went to the submodule directly to commit and push, but it returned "Error 403"

The screenshot showed the data that was not updated in "docs/demos/demos_conf.py"
Screenshot 2023-10-03 084108

**Review by hirwa-nshuti — APPROVED** [review]
Looks good to me 😊
Going to merge

**Pull request merged**